src/Orleans/Async/Cancellation/ICancellationSourcesExtension.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    internal interface ICancellationSourcesExtension : IGrainExtension, IGrain
+    {
+        [AlwaysInterleave]
+        Task CancelTokenSource(GrainCancellationToken token);
```

Comment:
No need to send the GrainCancellationToken . Just send Guid and what ever else you need.
That way it will be much cleaner to see what we need vs. half baked Token.
In the destination, we anyway lookup old Token by Guid.